contracts/Manager.sol

```diff
@@ -0,0 +1,413 @@
+
+    // Proposals
+    mapping(uint256 => Proposal) public proposals;
+    uint256 public proposalsLength;
+    uint256 public constant delay = 24 hours;
+    
+    // Issuance and Redemption controls
+    mapping(address => bool) public whitelist;
+    bool public useWhitelist;
+    bool public paused;
+
+    // The spread between issuance and redemption in BPS.
+    uint256 public seigniorage;                 // 0.1% spread -> 10 BPS
+    uint256 constant BPS_FACTOR = 10000;        // This is what 100% looks like in BPS.
+    
+
+    // EVENTS
+
+    event ProposalsCleared();
+
+    // RSV traded events
+    event Issuance(address indexed user, uint256 indexed amount);
+    event Redemption(address indexed user, uint256 indexed amount);
+
+    // Whitelist events
+    event WhitelistChanged(address indexed user, bool val);
+
+    // Pause events
+    event Paused(address indexed account);
+    event Unpaused(address indexed account);
+
+    // Changes
+    event OperatorChanged(address indexed account);
+    event SeigniorageChanged(uint256 oldVal, uint256 newVal);
+
+    // Proposals
+    event NewBasketProposalCreated(uint256 indexed id, address indexed proposer, address[] tokens, uint256[] backing);
+    event NewQuantityAdjustmentProposalCreated(uint256 indexed id, address indexed proposer, address[] tokens, uint256[] quantitiesIn, uint256[] quantitiesOut);
+    event ProposalAccepted(uint256 indexed id, address indexed proposer);
+    event ProposalCanceled(uint256 indexed id, address indexed proposer, address indexed canceler);
+    event ProposalExecuted(uint256 indexed id, address indexed proposer, address indexed executor);
+
+
+    // ============================ Constructor ===============================
+
+    /// Begins paused.
+    constructor(address vaultAddr, address rsvAddress, uint256 seigniorage_) public {
+        vault = IVault(vaultAddr);
+        rsv = IRSV(rsvAddress);
+        whitelist[_msgSender()] = true;
+        seigniorage = seigniorage_;
+        paused = true;
+        useWhitelist = true;
+    }
+
+    // ============================= Modifiers ================================
+
+    /// Modifies a function to run only when the contract is not paused.
+    modifier notPaused() {
+        _notPaused();
+        _;
+    }
+
+    /// Modifies a function to run only when the caller is on the whitelist, if it is enabled.
+    modifier onlyWhitelist() {
+        _onlyWhitelist();
+        _;
+    }
+
+    /// Modifies a function to run only when the caller is the operator account. 
+    modifier onlyOperator() {
+        _onlyOperator();
+        _;
+    }
+
+    function _notPaused() internal view {
+        require(!paused, "contract is paused");
+    }
+
+    function _onlyWhitelist() internal view {
+        if (useWhitelist) require(whitelist[_msgSender()], "not on whitelist");
+    }
+
+    function _onlyOperator() internal view {
+        require(_msgSender() == operator, "operator only");
+    }
```

Comment:
I would much prefer to have _notPaused, _onlyWhitelist, and _onlyOperator written inline in the modifiers above rather than be pulled out to separate functions like this, unless there is an important reason to do it this way.

Reply:
Writing it this way reduces bytecode, because the way modifiers actually work is they are in-lined in every place they are used.

Reply:
I see. Alas. Can you add a comment to that effect?

Reply:
It's inlined now.